nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/producer/FileBlobMessageProducer.java

```diff
@@ -39,50 +39,79 @@
 
     protected final File listFile;
 
-    protected int count = 0;
+    protected final String basePath;
 
-    protected Stream<String> stream;
+    protected long count = 0;
+
+    protected final long nbBlobs;
 
     protected Iterator<String> fileIterator;
 
-    public FileBlobMessageProducer(int producerId, File listFile) {
+    protected Stream<String> lines;
+
+    public FileBlobMessageProducer(int producerId, File listFile, String basePath, long nbBlobs) {
         super(producerId);
         this.listFile = listFile;
+        this.nbBlobs = nbBlobs;
+        if (basePath == null) {
+            this.basePath = "";
+        } else {
+            this.basePath = basePath;
```

Comment:
Remplace the if/then/else with `this.basePath = StringUtils.defaultString(basePath)`